airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -319,11 +320,38 @@ def post_dag_run(*, dag_id: str, session: Session = NEW_SESSION) -> APIResponse:
     if not dagrun_instance:
         try:
             dag = get_airflow_app().dag_bag.get_dag(dag_id)
+
+            data_interval_start_exists = (
+                "data_interval_start" in post_body and post_body["data_interval_start"] is not None
+            )
+            data_interval_end_exists = (
+                "data_interval_end" in post_body and post_body["data_interval_end"] is not None
+            )
```

Comment:
```suggestion
            data_interval_start_exists = post_body.get("data_interval_start") is not None
            data_interval_end_exists = post_body.get("data_interval_end") is not None
```
wdyt about using get here

Reply:
good point, done.